packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -361,21 +371,19 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
           return;
 
         case KeyCodes.tab:
-          if (this.props.allowTabKey) {
+          if (this.props.allowTabKey ||
+            this.props.handleTabKey === FocusZoneTabbableElements.all ||
+            (this.props.handleTabKey === FocusZoneTabbableElements.inputOnly && this._isElementInput(ev.target as HTMLElement))) {
+            let focusChanged = false;
+            this._processingTabKey = true;
```

Comment:
create a private member that's a FocusZoneTabbableElement type that gets updated on render and takes into account allowTabKey and then you can use that member instead of having to check both props